File: DevExpress-Demo/CS/MVVMDemo.Wpf/Modules/DXBinding/BindingExpressionsViewModel.cs
using System.Windows;

namespace MVVMDemo.DXBindingDemo {
    public class BindingExpressionsViewModel {
        public virtual string FirstName { get; set; }
        public virtual string LastName { get; set; }

        public virtual bool IsReadonly { get; set; }

        public virtual string UserName { get; set; }
        public virtual bool AcceptTerms{ get; set; }
        public void Register() {
            MessageBox.Show("Registered");
        }

        protected BindingExpressionsViewModel() {
            FirstName = "Gregory";
            LastName = "Price";
        }

        
    }
}

File: DevExpress-Demo/VB/MVVMDemo.Wpf/Modules/DXBinding/BindingExpressionsViewModel.vb
Imports System.Windows

Namespace MVVMDemo.DXBindingDemo

    Public Class BindingExpressionsViewModel

        Public Overridable Property FirstName As String

        Public Overridable Property LastName As String

        Public Overridable Property IsReadonly As Boolean

        Public Overridable Property UserName As String

        Public Overridable Property AcceptTerms As Boolean

        Public Sub Register()
            MessageBox.Show("Registered")
        End Sub

        Protected Sub New()
            FirstName = "Gregory"
            LastName = "Price"
        End Sub
    End Class
End Namespace
